controller/common/lut.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Union
+
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(
+        self,
+        lookup_table: Union[List[List[int | float]], np.ndarray],
+        interpolation_method: str = "linear",
+    ):
+        """
+        Initializes the LUT object.
+
+        Args:
+            lookup_table (list): A list of tuples or lists containing x-y data points for
+            interpolation. Format is [[x1,y1], [x2,y2], ..., [xn,yn]]
+            interpolation_method (str): Interpolation method to use. Default is "linear".
+
+        Raises:
+            ValueError: If the specified interpolation method is unknown.
+        """
+        if type(lookup_table) is np.ndarray:
+            self.__table = lookup_table
+        else:
+            self.__table = np.array(lookup_table)
```

Comment:
Unless you're using `self.__table` outside the constructor, I don't see the need to store it as a class member since you are using `self.x` and `self.y`

Reply:
so I made `self.__table` into a local variable `table`:
```
        if isinstance(lookup_table, NDArray):
            table = lookup_table
        else:
            table = np.array(lookup_table)

        self.__verifyTable(table)
        self.x = table[:, 0]
        self.y = table[:, 1]

```